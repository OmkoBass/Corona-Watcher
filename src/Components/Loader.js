import React from 'react'

//Bootstrap
import Spinner from "react-bootstrap/Spinner";

function Loader() {
    return (
        <div className='loading'>
            <Spinner animation="border" role="status">
                <span className="sr-only">Loading...</span>
            </Spinner>
        </div>
    )
}

export default Loader
