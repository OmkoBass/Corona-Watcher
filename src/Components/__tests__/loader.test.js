import React from "react";

import { render, screen, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

import Loader from "../Loader";

afterEach(() => {
  cleanup();
});

it("Should show loader", () => {
  render(<Loader />);

  const loaderElement = screen.getByTestId("loader");

  expect(loaderElement).toBeInTheDocument();
});

it("Should have Loading... as text content", () => {
  render(<Loader />);

  const loaderElement = screen.getByTestId("loader");

  expect(loaderElement).toHaveTextContent("Loading...");
});

it("Should contain a <span> tag", () => {
  render(<Loader />);

  const loaderElement = screen.getByTestId("loader");

  expect(loaderElement.firstChild).toHaveClass("spinner-border");
});

it("renders correctly", () => {
  const tree = renderer.create(<Loader />).toJSON();

  expect(tree).toMatchSnapshot();
});
