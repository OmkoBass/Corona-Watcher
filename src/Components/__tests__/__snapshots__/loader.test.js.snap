// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="loading"
  data-testid="loader"
>
  <div
    className="spinner-border"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
